rust/azure_iot_operations_connector/examples/base_connector_sample.rs

```diff
@@ -86,9 +86,95 @@ async fn run_program(
             }
         };
 
+        // Start handling the assets for this device endpoint
+        // if we didn't accept the inbound endpoint, then there's no reason to manage the assets
+        if endpoint_status.is_ok() {
+            tokio::task::spawn({
+                async move {
+                    run_assets(asset_creation_observation).await;
+                }
+            });
+        }
+
         device_endpoint_client
             .report_status(Ok(()), endpoint_status)
             .await;
     }
     panic!("device_creation_observer has been dropped");
 }
+
+// This function runs in a loop, waiting for asset creation notifications.
+async fn run_assets(mut asset_creation_observation: AssetClientCreationObservation) {
+    // Wait for a device creation notification
+    while let Some((asset_client, _asset_update_observation, _asset_deletion_token)) =
+        asset_creation_observation.recv_notification().await
+    {
+        log::info!("Asset created: {asset_client:?}");
+
+        // now we should update the status of the asset
+        let asset_status = match asset_client.specification().manufacturer.as_deref() {
+            Some("Contoso") | None => Ok(()),
+            Some(m) => {
+                log::warn!(
+                    "Asset '{}' not accepted. Manufacturer '{m}' not supported.",
+                    asset_client.asset_ref().name
+                );
+                Err(azure_device_registry::ConfigError {
+                    message: Some("asset manufacturer type is not supported".to_string()),
+                    ..azure_device_registry::ConfigError::default()
+                })
+            }
+        };
+
+        asset_client.report_status(asset_status).await;
+
+        for dataset in asset_client.datasets() {
+            tokio::task::spawn({
+                let dataset_clone = dataset.clone();
+                async move {
+                    run_dataset(dataset_clone).await;
+                }
+            });
+        }
+    }
+    panic!("asset_creation_observer has been dropped");
+}
+
+async fn run_dataset(dataset_client: DatasetClient) {
+    log::info!("new Dataset: {dataset_client:?}");
+
+    // now we should update the status of the dataset and report the message schema
+    dataset_client.report_status(Ok(())).await;
+    match dataset_client
+        .report_message_schema(
+            schema_registry::PutRequestBuilder::default()
+                .content(
+                    r#"
```

Comment:
Since the connector already uses `serde_json`, would the `json!` macro be useful here?

Reply:
probably, but since this is very brute force just to test for now, I'll leave it as is (if this were a better example, this json would actually get generated from the dataset). Ideally, this sample either won't exist very soon, or we'll use the actual implementation to generate this message schema